src/routes/router.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    FunctionComponent,
+    PropsWithChildren,
+    useEffect,
+    useMemo,
+    useState,
+} from 'react';
+import { FormattedMessage } from 'react-intl';
+import {
+    AuthenticationRouter,
+    getPreLoginPath,
+    initializeAuthenticationProd,
+} from '@gridsuite/commons-ui';
+import {
+    createBrowserRouter,
+    Navigate,
+    Outlet,
+    RouteObject,
+    RouterProvider,
+    useLocation,
+    useMatch,
+    useNavigate,
+} from 'react-router-dom';
+import { UserManager } from 'oidc-client';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppState } from '../redux/reducer';
+import { AppsMetadataSrv, UserAdminSrv } from '../services';
+import App from '../components/app';
+import { Connections, Users } from '../pages';
+import ErrorPage from './ErrorPage';
+import { updateUserManager_ } from '../redux/actions';
+import HomePage from './HomePage';
+
+export enum MainPaths {
+    users = 'users',
+    connections = 'connections',
+}
+
+export function appRoutes(): RouteObject[] {
```

Comment:
Why a function ?

Reply:
It's a structure config with components having dependencies in React context, so I can't per-iniitialize it in a constant outside.